symengine/tests/basic/test_functions.cpp

```diff
@@ -4062,3 +4063,17 @@ TEST_CASE("min: functions", "[functions]")
 
     CHECK_THROWS_AS(min({c}), SymEngineException);
 }
+
+TEST_CASE("test_dummy", "[Dummy]")
+{
+    RCP<const Symbol> x1 = symbol("x");
+    RCP<const Symbol> x2 = symbol("x");
+    RCP<const Symbol> xdummy1 = dummy("x");
+    RCP<const Symbol> xdummy2 = dummy("x");
+
+    CHECK(eq(*x1, *x2));
+    CHECK(neq(*x1, *xdummy1));
+    CHECK(neq(*xdummy1, *xdummy2));
+    CHECK(neq(*dummy(), *dummy()));
+    CHECK(neq(*dummy("x"), *dummy("x")));
```

Comment:
@isuruf I'm unable to understand why the above line passes while `4076` and `4077` fail.